dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -160,18 +104,60 @@ def __add__(self, other):
             )
         return merged_profile_compiler
 
+    
     def update_profile(self, df_series):
         """
-        Initializes the profiles the column dataframe.
+        Updates the profiles from the data frames
         
         :param df_series: a given column
         :type df_series: pandas.core.series.Series
-        :return: None
-        :rtype: None
+        :return: Self
+        :rtype: BaseColumnProfileCompiler
         """
+        
+        if len(self._profilers) == 0:
+            return 
+        
         df_series = df_series.apply(str)
-        for column_profile in self._profiles:
-            self._profiles[column_profile].update(df_series)            
+                
+        # If single process, loop and return
+        if not self.multiprocess_flag:
+            for col_profile in self._profiles:
+                self._profiles[col_profile].update(df_series)
+            return self
```

Comment:
this no longer is being caught by errors and specifying the `warn_on_profile`

Could we put hte other code in the if and then the code below isn't duplicated